core/player/src/view/plugins/template.ts

```diff
@@ -37,7 +39,12 @@ export type TemplateSubstitutionsFunc = (
 export default class TemplatePlugin implements ViewPlugin {
   private readonly options: Options;
 
-  hooks = {
+  hooks: {
+    resolveTemplateSubstitutions: SyncWaterfallHook<
+      [TemplateSubstitution[], TemplateItemInfo],
+      Record<string, any>
+    >;
+  } = {
```

Comment:
@KetanReddy is there a way we can turn off verbose for these? This kinds sucks to redeclare

Reply:
Yea VS Code was complaining about this; wasn't sure if I needed to do this or not.

Reply:
Unfortunately I think that comes from isolated declarations in tsc which we _should_ be using.

Reply:
I wonder if defining types for `hooks` means you could omit the type parameters from `resolveTemplateSubstitutions: new SyncWaterfallHook()` below